packages/common/src/objects.ts

```diff
@@ -58,6 +58,26 @@ export function mapKeys<T extends object, R extends string>(
     ) as Record<R, T[keyof T]>;
 }
 
+/**
+ * Checks if value is an object, e.g. a plain object, an array, a function,
+ * a regex, but not a primitive value.
+ *
+ * Common usage scenario:
+ * ```ts
+ * declare const value: unknown;
+ *
+ * // Type error: property 'foo' does not exist on type 'object'.
+ * typeof value === 'object' && value !== null && value.foo === 'bar';
+ *
+ *  // No type error.
+ * isObject(value) && value.foo === 'bar';
+ * ```
+ */
+export function isObject(value: unknown): value is Record<string | number | symbol, unknown> {
```

Comment:
`value is Record<string | number | symbol, unknown>` is pretty loose. it allows any other key to be added to the error if it passes for isObject.
perhaps `value is object`? this is where the `in` operator comes in handy:
`isObject(error) && 'name' in error  && 'message' in error && typeof error.name === 'string' && typeof error.message === 'string'`

this way we avoid the loose type

Reply:
> it allows any other key to be added to the error if it passes for isObject

Do you mean something like:
```ts
function isErrorLikeObject(error: unknown) {
    if (isObject(error)) {
        error.code = 5;
    }
}
```

It's not a very realistic scenario, `isObject` is mainly used to reduce line noise when validating unknown values, here's a real-world example:
```ts
isObject(error) && isObject(error.response) && isObject(error.response.data);
```
Equivalent example without using `isObject`:
```ts
typeof error === 'object' && error !== null && 'response' in error &&
typeof error.response === 'object' && error.response !== null &&
'data' in error.response && typeof error.response.data === 'object' &&
error.response.data !== null;
```

While `is object` assertion is more technically correct, it brings us closer to the second example. I can do `Readonly<Record>`. I'll do that.